lib/AST/ASTContext.cpp

```diff
@@ -8059,6 +8080,11 @@ QualType ASTContext::mergeFunctionTypes(QualType lhs, QualType rhs,
       }
     }
 
+    // Compatible functions must have the same number of type variables
+    // FIXME : Make sure to pass correct QualType.
```

Comment:
I don't understand the FIXME comment.  It doesn't look like there's anything to check.

I suggest moving this check earlier so that is with other similar checks such as checks on the number of parameters.